common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandResult.java

```diff
@@ -22,23 +22,49 @@
 //  THE SOFTWARE.
 package com.microsoft.identity.common.internal.controllers;
 
+import androidx.annotation.Nullable;
+
 public class CommandResult {
 
-    public enum ResultStatus { CANCEL, COMPLETED, ERROR };
+    public String getCorrelationId() {
+        return mCorrelationId;
+    }
+
+    public enum ResultStatus {
+        CANCEL,
+        COMPLETED,
+        ERROR;
+
+        public String getLogStatus() {
+            if (ResultStatus.COMPLETED == this) {
+                return "SUCCESS";
+            } else if (ResultStatus.ERROR == this) {
+                return "ERROR";
+            } else {
+                return "CANCEL";
+            }
+        }
+    }
 
     private ResultStatus mStatus;
     private Object mResult;
```

Comment:
It looks like these are effectively immutable already - we could make it official at the reference level by adding 'final'

Reply:
Not sure about immutability @AdamBJohnsonx since we do set correlationId on result if its a LocalAuthenticationResult

Reply:
Right, but we don't change the object reference in this class. final doesn't say anything about any references contained in the object, just the one in this class.  We don't reset it, so we could make it final.